lib/Model/ModelGraphPrinter.cpp

```diff
@@ -0,0 +1,235 @@
+                                  const model::Type *Parent,
+                                  uint64_t Offset,
+                                  uint64_t Index) {
+  // Specify port
+  Out << "<P" << to_string(Index) << ">";
+
+  // Dump the qualified type of the field
+  Out << FieldQT.UnqualifiedType.get()->name() << " ";
+  for (auto &Q : FieldQT.Qualifiers) {
+    switch (Q.Kind) {
+    case model::QualifierKind::Pointer:
+      Out << "*";
+      break;
+    case model::QualifierKind::Array:
+      Out << "[" << Q.Size << "]";
+      break;
+    default:
+      break;
+    }
+  }
+  Out << "\\n";
+
+  if (llvm::isa<model::StructType>(Parent))
+    Out << "\\nOffset: " << to_string(Offset);
+
+  if (FieldQT.size())
+    Out << "\\nSize: " << to_string(*FieldQT.size());
+}
+
+void ModelGraphPrinter::dumpNode(const model::Type *T, int NodeID) {
+
+  // Print the name of the type
+  Out << "node_" << to_string(NodeID) << "[";
+
+  // TODO: improve color coding
+  if (T->Kind == model::TypeKind::UnionType)
+    Out << "color=blue, ";
+  else if (T->Kind == model::TypeKind::CABIFunctionType)
+    Out << "color=green, ";
+  else if (T->Kind == model::TypeKind::RawFunctionType)
+    Out << "color=green, ";
+
+  Out << "label=\"{{" << T->name()
+      << ", size: " << to_string(T->size().value_or(0)) << "} ";
+
+  const FieldList &Fields = collectFields(T);
+
+  if (not Fields.empty()) {
+    Out << "|";
+    llvm::StringRef Sep = "";
+
+    bool DisplayVertically = DisplayUnionsVertically
+                             and T->Kind != model::TypeKind::UnionType;
+
+    if (not DisplayVertically)
+      Sep = "{";
+
+    // Append fields as part of the node's label
+    uint64_t CurOffset = 0;
+    for (auto Field : llvm::enumerate(Fields)) {
+      const auto &[FieldQT, Offset] = Field.value();
+      Out << " " << Sep << " ";
+
+      // Check if there's padding to be added
+      if (Offset > CurOffset) {
+        Out << " padding\\n(" << to_string(Offset - CurOffset) << ") | ";
+      }
+      CurOffset += FieldQT.size().value_or(0);
+
+      dumpField(FieldQT, T, Offset, Field.index());
+      Sep = "|";
+    }
+
+    if (not DisplayVertically)
+      Out << "}";
+  }
+
+  Out << "}\"];\n";
+}
+
+void ModelGraphPrinter::addEdge(const model::QualifiedType &QT,
+                                int SrcID,
+                                int SrcPort,
+                                int DstID) {
+  // Edge
+  Out << "node_" << to_string(SrcID) << ":<P" << to_string(SrcPort) << ">";
+  Out << " -> ";
+  Out << "node_" << to_string(DstID);
+
+  // Label
+  Out << "[label=\"";
+
+  for (auto &Qual : QT.Qualifiers) {
+    switch (Qual.Kind) {
+    case model::QualifierKind::Array:
+      Out << "Array[" << Qual.Size << "]";
+      break;
+
+    case model::QualifierKind::Pointer:
+      Out << "Pointer (size " << Qual.Size << ")";
+      break;
+
+    default:
+      break;
+    }
+
+    Out << ",\\n";
+  }
```

Comment:
There's a spurious comma at the end.
Something like:
```suggestion
  const char *Prefix = "";
  for (auto &Qual : QT.Qualifiers) {
    Out << Prefix;
    Prefix = ",\\n";
    switch (Qual.Kind) {
    case model::QualifierKind::Array:
      Out << "Array[" << Qual.Size << "]";
      break;

    case model::QualifierKind::Pointer:
      Out << "Pointer (size " << Qual.Size << ")";
      break;

    default:
      break;
    }
  }
```